integration/test/metric_value_benchmark/mem_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric_value_benchmark
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent/integration/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent/integration/test/status"
+	"time"
+)
+
+type MemTestRunner struct {
+}
+
+var _ ITestRunner = (*MemTestRunner)(nil)
+
+func (m *MemTestRunner) validate() status.TestGroupResult {
+	metricsToFetch := m.getMeasuredMetrics()
+	testResults := make([]status.TestResult, len(metricsToFetch))
+	for i, name := range metricsToFetch {
+		testResults[i] = m.validateMemMetric(name)
+	}
+
+	return status.TestGroupResult{
+		Name:        m.getTestName(),
+		TestResults: testResults,
+	}
+}
+
+func (m *MemTestRunner) getTestName() string {
+	return "Mem"
+}
+
+func (m *MemTestRunner) getAgentConfigFileName() string {
+	return "mem_config.json"
+}
+
+func (m *MemTestRunner) getAgentRunDuration() time.Duration {
+	return minimumAgentRuntime
+}
+
+func (m *MemTestRunner) getMeasuredMetrics() []string {
+	return []string{
+		"mem_active", "mem_available", "mem_available_percent", "mem_buffered", "mem_cached",
+		"mem_free", "mem_inactive", "mem_total", "mem_used", "mem_used_percent"}
+}
+
+func (m *MemTestRunner) validateMemMetric(metricName string) status.TestResult {
```

Comment:
This method feels like its gonna be the same for all plugins?

Reply:
I think a separate PR would be cleaner given this is mainly focused on the mem test.